processor.js

```diff
@@ -0,0 +1,70 @@
+'use strict';
+
+var doiuse = require('doiuse');
+var github = require('octonode');
+var postcss = require('postcss');
+var sourceMap = require('source-map');
+var stylus = require('stylus');
+var logger = require('./logger');
+
+function processCSS(repo, branch, file, config, handleIncompatibility) {
+    var repoClient = github.client().repo(repo);
+
+    repoClient.contents(file.filename, branch, function(error, fileContentsMetadata) {
+        var fileContents;
+
+        if (error) return logger.error('Error reading contents from ' + repo + ' / ' + branch + ' / ' + file.filename + ':', error);
+
+        fileContents = new Buffer(fileContentsMetadata.content, 'base64').toString();
+
+        postcss(doiuse({
+            browsers: config,
+            onFeatureUsage: handleIncompatibility
+        })).process(fileContents, {
+            from: '/' + file.filename
+        }).then(); // onFeatureUsage won't be called unless then() is called.
+    });
+}
+
+function processStylus(repo, branch, file, config, handleIncompatibility) {
+    var repoClient = github.client().repo(repo);
+
+    repoClient.contents(file.filename, branch, function(error, fileContentsMetadata) {
+        var fileContents, compiler;
+
+        if (error) return logger.error('Error reading contents from ' + repo + ' / ' + branch + ' / ' + file.filename + ':', error);
+
+        fileContents = new Buffer(fileContentsMetadata.content, 'base64').toString();
+        compiler = stylus(fileContents)
+            .set('filename', file.filename)
+            .set('sourcemap', {});
+
+        compiler.render(function(error, compiledCSS) {
+            var sourceMapConsumer;
+
+            if (error) return logger.error('Error compiling Stylus of ' + repo + ' / ' + branch + ' / ' + file.filename + ':', error);
+
+            sourceMapConsumer = new sourceMap.SourceMapConsumer(compiler.sourcemap);
```

Comment:
This and its initial var declaration can be moved into handleStylusIncompatibility